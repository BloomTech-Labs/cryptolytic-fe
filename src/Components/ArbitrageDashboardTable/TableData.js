import React from "react";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles({
  binanceColor: {
    color: "rgb(78, 185, 255)",
    padding: "0.5em 0",
    borderBottom: "1px solid rgba(35, 32, 44, 0.9)"
  },
  coinbaseColor: {
    color: "rgb(83, 207, 215)",
    padding: "0.5em 0",
    borderBottom: "1px solid rgba(35, 32, 44, 0.9)"
  },
  bitfinex: {
    color: "rgb(98, 227, 171)",
    padding: "0.5em 0",
    borderBottom: "1px solid rgba(35, 32, 44, 0.9)"
  },
  tradingPair: {
    display: "flex",
    alignItems: "center",
    paddingLeft: "23%",
    padding: "0.5em 0",
    borderBottom: "1px solid rgba(35, 32, 44, 0.9)",

    "@media(max-width: 1600px)": {
      padding: "1.2em"
    }
  },
  data: {
    padding: "0.5em 0",
    borderBottom: "1px solid rgba(35, 32, 44, 0.9)"
  }
});

const TableData = ({ item, positive, negative, icon1, icon2 }) => {
  const classes = useStyles();

  if (positive && item >= 0) {
    return <td className={positive}>{item + " %"}</td>;
  } else if (negative && negative) {
    return <td className={negative}>{item + " %"}</td>;
  } else if (item === "Binance") {
    return <td className={classes.binanceColor}>{item}</td>;
  } else if (item === "Coinbase Pro") {
    return <td className={classes.coinbaseColor}>{item}</td>;
  } else if (item === "Bitfinex") {
    return <td className={classes.bitfinex}>{item}</td>;
  } else if (icon1 && icon2) {
    return (
      <td className={classes.tradingPair}>
        <img src={icon1} />
        <img src={icon2} />

        {item}
      </td>
    );
  } else {
    return <td className={classes.data}>{item}</td>;
  }
};

export default TableData;
