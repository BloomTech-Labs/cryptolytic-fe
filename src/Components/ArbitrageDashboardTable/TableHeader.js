import React from "react";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles({
  headings: {
    padding: "0.5em 0",
    position: "sticky",
    top: 0,
    background: "#191E23",

    "@media(max-width: 1600px)": {
      padding: "1em"
    }
  }
});

const TableHeader = ({ header }) => {
  const classes = useStyles();

  header = header
    .split(" ")
    .map(word => {
      return word.charAt(0).toUpperCase() + word.substr(1);
    })
    .join(" ");

  return <th className={classes.headings}>{header}</th>;
};

export default TableHeader;
