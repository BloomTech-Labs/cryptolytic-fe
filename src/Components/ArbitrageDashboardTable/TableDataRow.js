import React from "react";
import TableData from "./TableData";
import { getIcons } from "../../Utilities/tradingPairIcons";

import { makeStyles } from "@material-ui/core/styles";
import TableRow from "@material-ui/core/TableRow";

const useStyles = makeStyles({
  tablerow: {
    background: "rgba(0, 0, 0, 0.8)",
    color: "rgb(255, 255, 255)"
  },
  positive: {
    color: "green",
    padding: "0.5em 0",
    borderBottom: "1px solid rgba(35, 32, 44, 0.9)"
  },
  negative: {
    color: "red",
    padding: "0.5em 0",
    borderBottom: "1px solid rgba(35, 32, 44, 0.9)"
  }
});

const TableDataRow = ({ dataRow }) => {
  console.log(dataRow, `arb`);
  const classes = useStyles();

  dataRow.buy_exchange = dataRow.buy_exchange
    .replace(/\_/, " ")
    .split(" ")
    .map(word => {
      return word.charAt(0).toUpperCase() + word.substr(1);
    })
    .join(" ");

  dataRow.sell_exchange = dataRow.sell_exchange
    .replace(/\_/, " ")
    .split(" ")
    .map(word => {
      return word.charAt(0).toUpperCase() + word.substr(1);
    })
    .join(" ");

  dataRow.trading_pair = dataRow.trading_pair.toUpperCase();
  dataRow.trading_pair = dataRow.trading_pair.replace(/\_/, "/");

  if (String(dataRow.price_difference).length <= 3) {
    dataRow.price_difference =
      "$ " + String(dataRow.price_difference.toFixed(2));
  } else {
    dataRow.price_difference = "$ " + String(dataRow.price_difference);
  }

  if (String(dataRow.arbritage_percentage).length <= 3) {
    dataRow.arbritage_percentage = dataRow.arbritage_percentage.toFixed(2);
  }

  const data = Object.values(dataRow);

  const icons = getIcons(dataRow.trading_pair);
  return (
    <TableRow className={classes.tablerow}>
      {data.map((item, i) => {
        if (i === 6) {
          return (
            <TableData
              item={item}
              positive={classes.positive}
              negative={classes.negative}
            />
          );
        } else if (i === 4) {
          return (
            <TableData item={item} icon1={icons.icon1} icon2={icons.icon2} />
          );
        }
        return <TableData item={item} />;
      })}
    </TableRow>
  );
};

export default TableDataRow;
