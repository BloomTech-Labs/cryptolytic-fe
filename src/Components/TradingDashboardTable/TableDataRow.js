import React from "react";
import TableData from "./TableData";
import { getIcons } from "../../Utilities/tradingPairIcons";

import { withStyles, makeStyles } from "@material-ui/core/styles";
import TableRow from "@material-ui/core/TableRow";
import { ListItemIcon } from "@material-ui/core";

const useStyles = makeStyles({
  tablerow: {
    background: "rgba(0, 0, 0, 0.8)",
    color: "rgb(255, 255, 255)",
    borderBottom: "1px solid white"
  },
  positive: {
    color: "green",
    padding: "0.5em 0",
    borderBottom: "1px solid rgba(35, 32, 44, 0.9)"
  },
  negative: {
    color: "red",
    padding: "0.5em 0",
    borderBottom: "1px solid rgba(35, 32, 44, 0.9)",
    color: "red"
  },
  alined: {
    textAlign: "left"
  }
});

const TableDataRow = ({ dataRow }) => {
  const classes = useStyles();
  console.log(dataRow, `trading`);

  dataRow.exchange = dataRow.exchange
    .replace(/\_/, " ")
    .split(" ")
    .map(word => {
      return word.charAt(0).toUpperCase() + word.substr(1);
    })
    .join(" ");

  dataRow.trading_pair = dataRow.trading_pair.toUpperCase();
  dataRow.trading_pair = dataRow.trading_pair.replace(/\_/, "/");

  if (String(dataRow.trade_price).length <= 3) {
    dataRow.trade_price = "$ " + String(dataRow.trade_price.toFixed(2));
  } else {
    dataRow.trade_price = "$ " + String(dataRow.trade_price);
  }

  if (String(dataRow.percentage).length <= 3) {
    dataRow.percentage = dataRow.percentage.toFixed(2);
  }

  const data = Object.values(dataRow);

  const icons = getIcons(dataRow.trading_pair);

  return (
    <TableRow className={classes.tablerow}>
      {data.map((item, i) => {
        if (i === 6) {
          return (
            <TableData
              item={item}
              positive={classes.positive}
              negative={classes.negative}
            />
          );
        } else if (i === 4) {
          return (
            <TableData item={item} icon1={icons.icon1} icon2={icons.icon2} />
          );
        }
        return <TableData item={item} alined={classes.alined} />;
      })}
    </TableRow>
  );
};

export default TableDataRow;
