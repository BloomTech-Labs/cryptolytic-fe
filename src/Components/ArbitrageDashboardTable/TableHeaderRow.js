import React from "react";
import TableHeader from "./TableHeader";

import { makeStyles } from "@material-ui/core/styles";
import TableRow from "@material-ui/core/TableRow";

const useStyles = makeStyles({
  tablerow: {
    background: "#23282D",
    color: "rgb(112, 112, 112)"
  }
});

const TableHeaderRow = ({ headerData }) => {
  headerData = headerData.map(header => {
    return header.replace(/_/, " ");
  });

  const classes = useStyles();

  return (
    <TableRow className={classes.tablerow}>
      {headerData.map(header => (
        <TableHeader header={header} />
      ))}
    </TableRow>
  );
};

export default TableHeaderRow;
