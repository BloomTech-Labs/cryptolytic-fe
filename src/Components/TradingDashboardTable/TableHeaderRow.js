import React from "react";
import TableHeader from "./TableHeader";

import { withStyles, makeStyles } from "@material-ui/core/styles";
import TableRow from "@material-ui/core/TableRow";
import zIndex from "@material-ui/core/styles/zIndex";

const useStyles = makeStyles({
  tablerow: {
    background: "#23282D",
    color: "rgb(112, 112, 112)"
  }
});

const TableHeaderRow = ({ headerData }) => {
  const classes = useStyles();

  headerData = headerData.map(header => {
    return header.charAt(0).toUpperCase() + header.substr(1);
  });

  return (
    <TableRow className={classes.tablerow}>
      {headerData.map(header => (
        <TableHeader className={classes.tableHead} header={header} />
      ))}
    </TableRow>
  );
};

export default TableHeaderRow;
