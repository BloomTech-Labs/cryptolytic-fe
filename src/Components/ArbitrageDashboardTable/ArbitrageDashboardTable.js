import React, { useState, useEffect } from "react";
import TableHeaderRow from "./TableHeaderRow";
import TableDataRow from "./TableDataRow";
import { css } from "@emotion/core";
import PropagateLoader from "react-spinners/PropagateLoader";
import axios from "axios";
import { makeStyles } from "@material-ui/core/styles";
import TableContainer from "@material-ui/core/TableContainer";
import Table from "@material-ui/core/Table";
import TableHead from "@material-ui/core/TableHead";
import TableBody from "@material-ui/core/TableBody";

const useStyles = makeStyles({
	table: {
		marginLeft: "20vw",
		width: "70%",
		background: "rgba(35, 32, 44, 0.8)",
		boxShadow: "0px 4px 4px rgba(0, 0, 0, 0.25)",
		"&::-webkit-scrollbar-thumb": {
			backgroundColor: "rgba(255,255,255,1.0)",
			outline: "1px solid slategrey"
		}
	},
	tableContainer: {
		maxHeight: 800,

		"&::-webkit-scrollbar": {
			display: "none"
		}
	}
});


const override = css`
  display: flex;
  justify-content: center;
  position: relative;
  margin: 30vh 0 50vh 10vw;
`;

const ArbitrageDashboardTable = () => {
	const [data, setData] = useState([]);
	const [loading, setLoading] = useState(true);

	useEffect(() => {
		axios
			.get("https://cryptolytic-starter.herokuapp.com/arbitrage")
			.then(res => {
				setData(res.data);
				setTimeout(() => {
					setLoading(false);
				}, 2000);
			})
			.catch(err => console.log(err.response));
	}, []);

	const arbitrageData = data.map(obj => {
		let {
			buy_exchange,
			sell_exchange,
			trading_pair,
			arbitrage_percentage,
			trade_time,
			price_difference
		} = obj;

		let date, time;
		trade_time = new Date(obj.trade_time * 1000);

		date = trade_time.toLocaleDateString();
		time = trade_time.toLocaleTimeString();

		return {
			date,
			time,
			buy_exchange,
			sell_exchange,
			trading_pair,
			price_difference,
			arbitrage_percentage
		};
	});

	const headerData = [];

	for (let key in arbitrageData[0]) {
		headerData.push(key);
	}

	const classes = useStyles();


	if (loading) {
		return (
			<PropagateLoader
				css={override}
				size={25}
				color={
					"linear-gradient(93.16deg, #4EB9FF 19.25%, #53CFD7 45.13%, #5DDCB7 67.95%, #62E3AB 82.93%);"
				}
				loading={loading}
			/>
		);
	} else {

		return (
			<TableContainer className={classes.tableContainer}>
				<Table className={classes.table}>
					<TableHead>
						<TableHeaderRow headerData={headerData} />
					</TableHead>
					<TableBody>
						{arbitrageData.map(dataRow => (
							<TableDataRow dataRow={dataRow} />
						))}
					</TableBody>
				</Table>
			</TableContainer>
		);
	};
};

export default ArbitrageDashboardTable;
