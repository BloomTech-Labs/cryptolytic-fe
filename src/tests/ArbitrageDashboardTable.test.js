import React from "react";
import { MemoryRouter } from "react-router";
///
import ArbitrageDashboardTable from "../Components/ArbitrageDashboardTable/ArbitrageDashboardTable";
import TableHeaderRow from "../Components/ArbitrageDashboardTable/TableHeaderRow";
import TableDataRow from "../Components/ArbitrageDashboardTable/TableDataRow";
///
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

test("ArbitrageDashboardTable components initialized without crashing", () => {
  render(<TableHeaderRow headerData={["Header", "Test"]} />);
  render(
    <TableDataRow
      dataRow={
        ({
          buy_exchange: "bitfinex",
          sell_exchange: "binance",
          trading_pair: "BTC_USD"
        },
        {
          buy_exchange: "binance",
          sell_exchange: "bitfinex",
          trading_pair: "ETH_BTC"
        })
      }
    />
  );
});

describe("<ArbitrageDashboardTable /> matches verified SnapShot", () => {
  it("should match snapshot", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <ArbitrageDashboardTable />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
