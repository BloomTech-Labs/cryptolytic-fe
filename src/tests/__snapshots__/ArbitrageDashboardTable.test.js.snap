// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArbitrageDashboardTable /> matches verified SnapShot should match snapshot 1`] = `
<div
  className="css-1n707d5"
>
  <div
    className="css-wrax94"
  />
  <div
    className="css-cxnvq8"
  />
  <div
    className="css-xy7ez6"
  />
  <div
    className="css-14panbf"
  />
  <div
    className="css-uuu0ai"
  />
  <div
    className="css-1svdgyg"
  />
</div>
`;
